libraries/AP_BoardConfig/board_drivers.cpp

```diff
@@ -199,11 +285,34 @@ void AP_BoardConfig::validate_board_type(void)
 #endif
 }
 
+
+
 /*
   auto-detect board type
  */
 void AP_BoardConfig::board_autodetect(void)
 {
+#if defined(HAL_CHIBIOS_ARCH_CUBEBLACK)
+    {
+        if (state.board_type != PX4_BOARD_PIXHAWK2) {
+            state.board_type.set(PX4_BOARD_PIXHAWK2);
+        }
+
+        bool success = true;
+        if (!spi_check_register("mpu9250", MPUREG_WHOAMI, MPU_WHOAMI_MPU9250)) { success = false; }
```

Comment:
I'm okay with it in this form, but ideally I think we'd provide this via a set of entries in the HWDEF.

Reply:
Agreed.